DocsScraper/src/DocsScraper.jl

```diff
@@ -0,0 +1,235 @@
+
+    ## return statements
+    # `is_code` specifies if the block is code or not
+    is_code = false
+
+    # `is_text_inserted` is a flag which specifies if the text is inserted in the subsequent recursion iteration. 
+    is_text_inserted = false
+
+    # process headings. Working: If the current tag is a heading, then, we remove all the smaller headings we stored (for metadata) from  
+    # heading_hierarchy dictionary and add the current heading tag
+    # This is done because, in the heading hierarchy, if we come back to the current heading, 
+    # then there is no need to store the smaller headings we encountered earlier
+    if startswith(tag_name, "h") && isdigit(last(tag_name))
+
+        # Clear headings of equal or lower level
+        for k in collect(keys(heading_hierarchy))
+            if Base.parse(Int, last(k)) >= Base.parse(Int, last(tag_name))
+                delete!(heading_hierarchy, k)
+            end
+        end
+        heading_hierarchy[tag_name] = strip(Gumbo.text(node))
+        push!(parsed_text,
+            Dict("heading" => strip(Gumbo.text(node)),
+                "metadata" => copy(heading_hierarchy)))
+        return "", is_code, is_text_inserted
+
+        # if the current tag is <code>, then get it's textual value and return it with backticks (``)
+    elseif tag_name == "code"
+        # Start a new code block
+        code_content = "`" * strip(Gumbo.text(node)) * "`"
+        is_code = true
+        return code_content, is_code, is_text_inserted
+    else
+        text_to_insert = ""
+
+        # Recursively process the child node for text content
+        children = collect(AbstractTrees.children(node))
+        num_children = length(children)
+
+        for (index, child) in enumerate(children)
+
+            # if the current tag belongs in the list, it is assumed that all the text/code should be part of a single paragraph/block, unless,
+            # there occurs a code block with >50 chars, then, previously parsed text is inserted first, then the code block is inserted. 
+            if tag_name in ["p", "li", "dt", "dd", "pre", "b", "strong", "i", "cite", "address", "em", "td"]
+                recieved_text, is_code, is_text_inserted = process_node!(child, heading_hierarchy, parsed_text, false, prev_text * " " * text_to_insert)
+            else
+                recieved_text, is_code, is_text_inserted = process_node!(child, heading_hierarchy, parsed_text, child_new, prev_text * " " * text_to_insert)
+            end
+
+            # changing `text_to_insert` to "" to avoid inserting `text_to_insert` again (as it was inserted by the child recursion call)
+            if is_text_inserted
+                text_to_insert = ""
+                prev_text = ""
+            end
+
+            # to separate the code into serparate blocks if the code is large and part of text on the webpage
+            if !isempty(strip(recieved_text)) && is_code == true && length(recieved_text) > 50
```

Comment:
Minor comment - I've already marked this in the past. Could you please remove the typo `recieved_*`?